garak/probes/suffix.py

```diff
@@ -108,3 +109,67 @@ def probe(self, generator) -> List[garak.attempt.Attempt]:
             logging.warning("GCG failed to find a jailbreak!")
             # Return an empty list to avoid throwing assertion errors upstream
             return list()
+
+
+class BEAST(Probe):
+    """
+    BEAST probe
+
+    Beam Search-based Adversarial attack.
+    Based on "Fast Adversarial Attacks in One GPU Minute" by Sadasivan et al. (https://arxiv.org/abs/2402.15570)
+    """
+
+    bcp47 = "en"
+    recommended_detector = ["mitigation.MitigationBypass"]
+    tags = [
+        "avid-effect:security:S0403",
+        "avid-effect:performance:P0204",
+        "quality:Security:PromptStability",
+        "payload:jailbreak",
+    ]
+    goal = "cause the model to disregard the system prompt"
+    doc_uri = "https://arxiv.org/abs/2402.15570"
+
+    prompts = list()
+    active = False
+
+    def __init__(self, goal: str = GOAL, config_root=_config):
+        self.goal = goal
+        self.run_beast = run_beast
+        super().__init__(config_root=config_root)
+
+    def probe(self, generator) -> List[garak.attempt.Attempt]:
+        self.generator = generator
+
+        try:
+            beast_output = self.run_beast(target_generator=self.generator)
+        except Exception as e:
```

Comment:
Love that there is error handling here.

Can this be more specific in the `Exception` types captured? Consider using `GarakException` and having any generated exceptions in the utility inherit.